### DISCUSSION TITLE
Jquery with next js

### DISCUSSION THREAD

**Discussion by Lolly1150** [vercel/next.js#36687] [Help] [answered]
I try to use jquery it is my assets javascript file (It only works when I edit the script at that moment and as if it wasn't being read the other times)
Code:
```js
$('.sidebarCollapse').on('click', function () {
	//my code
});

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Do you have more code to look at? Possibly a repository?


I wonder if your script is running before there's a `.sidebarCollapse`. Assuming it is not present in the initial HTML. 

More context is needed to help you out.

**Reply by Lolly1150** [reply]
On click element it don't execute code

**Reply by icyJoseph** [reply]
Well, I just setup `/my-script.js` in the public folder, and loaded `jQuery` from a CDN on my `_document` file, then input `/my-script.js` to `_app` using the next/script, and things work fine. 

Do you have a repository to take a look at?

**Reply by Lolly1150** [reply]
I load jquery in my script file using npm

**Reply by icyJoseph** [reply]
Could you make a small reproducible demo, using codesandbox? https://codesandbox.io/s/nextjs-5tlgz

**Reply by Lolly1150** [reply]
And as if sometimes jquery doesn't detect the click
My code is:
```jsx
<a className="sidebarCollapse" data-placement="bottom">
<svg
xmlns="http://www.w3.org/2000/svg"
width="24"
height="24"
viewBox="0 0 24 24"
fill="none"
stroke="currentColor"
strokeWidth="2"
strokeLinecap="round"
strokeLinejoin="round"
className="feather feather-menu"
>
<line x1="3" y1="12" x2="21" y2="12"></line>
<line x1="3" y1="6" x2="21" y2="6"></line>
<line x1="3" y1="18" x2="21" y2="18"></line>
</svg>
</a>

```

```js
$('.sidebarCollapse').on('click', function (sidebar) {
sidebar.preventDefault();
$(Selector.mainContainer).toggleClass('sidebar-closed');
$(Selector.mainHeader).toggleClass('expand-header');
$(Selector.mainContainer).toggleClass('sbar-open');
$('.overlay').toggleClass('show');
$('html,body').toggleClass('sidebar-noneoverflow');
});
```

**Reply by icyJoseph** [reply]
And how do you make jQuery available? Also please include the path to the file that makes jQuery available. If you have a private repository, you could add me and I can comment back here what I think the issue is.

**Reply by Lolly1150** [reply]
> E come rendi disponibile jQuery? Si prega inoltre di includere il percorso del file che rende disponibile jQuery. Se hai un repository privato, potresti aggiungermi e posso commentare qui quale penso sia il problema.

Invited

**Reply by icyJoseph** [reply]
Alright, I think the issue is that the `isLoading` flag is false and under before it becomes true, the `App.init` is invoked.

If you do add this to the Servers component:

```js
// pages/dashboard/servers.jsx
useEffect(() => {
     if (!isLoading) App.init();
}, [isLoading]);
```

Could you confirm?

**Reply by Lolly1150** [reply]
I do this for all the files where I have to initialize it right?
And one more thing can I add you to my credits page?

**Reply by icyJoseph** [reply]
Yes, you'd need to do this every time the `.sidebarCollapse` classname holder has been removed and reinserted from the DOM.

Sure, that'd be okay. Don't forget to mark as answered :).

I'd say that you should try to migrate away from jQuery though, it is difficult to run a project where two frameworks manipulate the DOM.
